### PR TITLE
Integration PR for #6541

### PR DESCRIPTION
**Pull request by matthiasb85**
Allow admin users to set a repositories visibility to public, even if FORCE_PRIVATE is to true (#6541)

Signed-off-by: Matthias Beckert <beckert.matthias@googlemail.com>

### CODE REVIEW

**Review by zeripath — APPROVED** [review]
Although usually I'm against duplication I think this looks cleaner than a 3 way "and".

Of note this allows users to set their repositories to private again if they've been made non-private by an admin but that's probably reasonable.

**Comment by matthiasb85** [comment]
You are right, the user can always switch back to private. Nevertheless, the switch from private to public can only be performed by an administrator (in case of FORCE_PRIVAT).  This is also the case, if the visibility is changed multiple times.

File: templates/repo/settings/options.tmpl [hunk 1]
```diff
@@ -19,7 +19,11 @@
 					<div class="inline field">
 						<label>{{.i18n.Tr "repo.visibility"}}</label>
 						<div class="ui checkbox">
+							{{if .IsAdmin}}
+							<input name="private" type="checkbox" {{if .Repository.IsPrivate}}checked{{end}}>
+							{{else}}
 							<input name="private" type="checkbox" {{if .Repository.IsPrivate}}checked{{end}}{{if and $.ForcePrivate .Repository.IsPrivate}} readonly{{end}}>
```
Comment (sapk) [inline templates/repo/settings/options.tmpl:25]: To follow the same check/logic as in `routers/repo/setting.go` you just need to add (not .IsAdmin) to the and list.
```suggestion
							<input name="private" type="checkbox" {{if .Repository.IsPrivate}}checked{{end}}{{if and $.ForcePrivate .Repository.IsPrivate (not .IsAdmin)}} readonly{{end}}>
```

**Review by sapk — CHANGES_REQUESTED** [review]

File: templates/repo/settings/options.tmpl (hunk 1 above)
Comment (sapk) [reply templates/repo/settings/options.tmpl:25]: I have just seen the comment of @zeripath and I will LGTM as it is not blocking.

**Review by sapk — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**